src/smpp_socket.erl

```diff
@@ -471,10 +475,15 @@ handle_pkt(#pdu{command_id = CmdID, sequence_number = Seq} = Pkt,
                             rate_limit_cooldown -> StateName;
                             _ -> bound
                         end,
-            {ok, NextState, State1};
+            State2 = case State1 of
+                #{in_flight := []} -> maps:without([response_time], State1);
+                #{in_flight := _} -> set_request_timeout(State1)
+            end,
+            {ok, NextState, State2};
         false ->
             report_unexpected_response(Pkt, StateName, State),
-            {ok, StateName, State}
+            State2 = set_keepalive_timeout(State, esme),
```

Comment:
Is it really needed here: on _incoming_ packet?

Reply:
WIP artifact, removed.